src/sortStudents.ts

```diff
@@ -1,16 +1,49 @@
 
 export interface Student {
-  // describe Student interface
+  name: 'string',
+  surname: 'string',
+  age: number,
+  married: boolean,
+  grades: [],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(arrStudent: number[]): number {
+  return arrStudent.reduce((a: number, b: number) => a + b) / arrStudent.length;
+}
+
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const arrStudent: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Age:
+      return arrStudent.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.age - b.age
+        : b.age - a.age));
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    case SortType.Name:
+    case SortType.Surname:
+      return arrStudent.sort((a: Student, b: Student) => (order === 'asc'
+        ? a[sortBy].localeCompare(b[sortBy])
+        : b[sortBy].localeCompare(a[sortBy])));
+    case SortType.Married:
+      return arrStudent.sort((a: Student, b: Student) => (order === 'asc'
```

Comment:
Можна не писати явно типи, TS і так це зрозуміє))
```suggestion
      return arrStudent.sort((a, b) => (order === 'asc'
```